src/view_control.cpp

```diff
@@ -1170,13 +1170,24 @@ void View::DrawSylConnector(
     if (!syl->GetStart() || !syl->GetEnd()) return;
 
     int y = staff->GetDrawingY() + GetSylYRel(syl->m_drawingVerse, staff);
-    TextExtend extend;
+
+    // Invalid bounding boxes might occur for empty syllables without text child
+    if (!syl->HasContentHorizontalBB()) return;
+
+    // Define lambda returning the next syllable with valid bounding box
+    auto nextValidSyl = [](Syl *syl) {
+        Syl *next = syl->m_nextWordSyl;
+        while (next && !next->HasContentHorizontalBB()) {
+            next = next->m_nextWordSyl;
+        }
+        return next;
+    };
 
     // The both correspond to the current system, which means no system break in-between (simple case)
     if (spanningType == SPANNING_START_END) {
         x1 = syl->GetContentRight();
-        if (syl->m_nextWordSyl) {
-            x2 = syl->m_nextWordSyl->GetContentLeft();
+        if (Syl *nextSyl = nextValidSyl(syl); nextSyl) {
```

Comment:
This is over the top in my opinion. Doing
```
if (syl->m_nextWordSyl) {
    x2 = (syl->m_nextWordSyl->HasContentHorizontalBB()) ? syl->m_nextWordSyl->GetContentLeft() : syl->m_nextWordSyl->GetDrrawingX();
}
```
would be fine.

The `while` loop in the lambda scares me because of the `next = next->m_nextWordSyl;` when trying to deal with the situation where something is not right.

Reply:
This would access the X drawing position of an empty syllable which seems to be not right. In the Foo-Bar example I would get (x1, x2)=(722, 90) instead of (722, 780). The output would contain no dash at all between _Foo_ and _Bar_.

What exactly scares you in the `while`-loop? The next pointer should always be NULL or point to another syllable. It is initialised with `NULL` and set in `Syl::PrepareLyrics(...)`. I could check for cycles. Or I could use a max bound on iterations to be safe, but that would look awkward as well.

Reply:
Yes, I understand what it does. But we are in a situation where we are dealing with data that does not make much sense and should be fixed. What is the point of having an empty syllable? And what is the point of trying to show connectors of empty syllables? 

The `while` loop is fine as long as the pointer is properly initialized. It should always be. I am thinking about having this when the edit function is being used. Of course, the pointer should be reset and it should be fine too. But again, considering what we are dealing with here I don't see the point of trying to get the next syllable when the data makes no sense. It is only a question of priorities.

Reply:
But then we could also exit in this case at the begin of the function. Maybe that is what you meant in the first place. I have changed it to that.

Reply:
Sorry, yes, this is what I meant. Thanks